src/extend/diastereotopic/getGroupedDiastereotopicAtomIDs.js

```diff
@@ -23,6 +25,47 @@ module.exports = function getGroupedDiastereotopicAtomIDs(options = {}) {
       }
     }
   }
+  // Find the Magnetically equivalent groups
+  let pathOptions = { maxLength: 9999 };
+  if (label) {
+    pathOptions.fromLabel = label;
+    pathOptions.toLabel = label;
+  }
+  let paths = this.getAllPaths(pathOptions);
+  for (let key of Object.keys(diaIDsObject)) {
+    let hashTable = {};
+    for (let i = 0; i < paths.length; i++) {
+      let pair = paths[i];
+      if (pair.fromDiaID === key && pair.toDiaID !== key) {
+        let groupB = pair.fromAtoms;
+        groupB.forEach((value, index) => {
+          if (!hashTable[value]) {
+            hashTable[value] = {};
+          }
+          hashTable[value][pair.toAtoms[index]] = pair.pathLength;
+        });
+      }
+    }
+    // console.log(hashTable);
+    let keys = Object.keys(hashTable);
+    let groups = {};
+    for (let atomID of keys) {
+      let uniqueColumn = JSON.stringify(hashTable[atomID]);
```

Comment:
Can you avoid JSON.stringify ? Maybe a Map ?

Reply:
It is an object. It is not an array and what I need is exactly the string representations of it. It is my hash key.

Reply:
`Map` is the builtin class to create hash tables with objects as keys: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map

Reply:
Actually none of them work. I need to build a key base on the key/value pairs, no matter the order.  So I'll need to sort the keys and then cat the key:values pairs